test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs

```diff
@@ -8546,6 +8550,418 @@ public async Task RestoreNetCore_ProjectProvidedRuntimeIdentifierGraphChange_Doe
             }
         }
 
+        [Theory]
+        [InlineData(new string[] { "win7-x86" }, new string[] { "win-x64" })]
+        [InlineData(new string[] { "win7-x86", "win-x64" }, new string[] { "win-x64" })]
+        [InlineData(new string[] { "win7-x86" }, new string[] { "win7-x86", "win-x64" })]
+        public void RestoreNetCore_PackagesLockFile_WithProjectChangeRuntimeAndLockedMode_FailsRestore(string[] intitialRuntimes, string[] updatedRuntimes)
+        {
+            // A project with RestoreLockedMode should fail restore if the project's runtime is changed between restores.
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+                var tfm = "net45";
+
+                var projectA = SimpleTestProjectContext.CreateNETCore(
+                   "a",
+                   pathContext.SolutionRoot,
+                   new NuGetFramework[]{ NuGetFramework.Parse(tfm) });
+
+                projectA.Properties.Add("RestorePackagesWithLockFile", "true");
+                projectA.Properties.Add("RestoreLockedMode", "true");
```

Comment:
As discussed, I think it makes sense to do this after the original restore.

Otherwise the fixing of the other bug that was created as a result of the discussion.

Reply:
Updated